providers/go-feature-flag/src/test/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProviderTest.java

```diff
@@ -269,307 +308,417 @@ void should_resolve_a_valid_boolean_flag_with_TARGETING_MATCH_reason_cache_disab
+                        Arrays.asList(new Value("test"),
+                                new Value("test1"),
+                                new Value("test2"),
+                                new Value("false"),
+                                new Value("test3")))))
+                .reason(Reason.TARGETING_MATCH.name())
+                .variant("True")
+                .flagMetadata(defaultMetadata)
+                .flagKey("list_key")
+                .build();
+        assertEquals(want, got);
     }
 
     @SneakyThrows
     @Test
     void should_not_fail_if_receive_an_unknown_field_in_response() {
         GoFeatureFlagProvider g = new GoFeatureFlagProvider(GoFeatureFlagProviderOptions.builder().endpoint(this.baseUrl.toString()).timeout(1000).build());
-        g.initialize(new ImmutableContext());
-        ProviderEvaluation<Boolean> res = g.getBooleanEvaluation("unknown_field", false, this.evaluationContext);
-        assertEquals(true, res.getValue());
-        assertNull(res.getErrorCode());
-        assertEquals(Reason.TARGETING_MATCH.toString(), res.getReason());
-        assertEquals("True", res.getVariant());
-        assertAll("Test flagMetadata",
-                () -> assertEquals(defaultMetadata.getInteger("version"), res.getFlagMetadata().getInteger("version")),
-                () -> assertEquals(defaultMetadata.getString("pr_link"), res.getFlagMetadata().getString("pr_link")));
+        String providerName = this.testName;
+        OpenFeatureAPI.getInstance().setProviderAndWait(providerName, g);
+        Client client = OpenFeatureAPI.getInstance().getClient(providerName);
+        FlagEvaluationDetails<Boolean> got = client.getBooleanDetails("unknown_field",false, this.evaluationContext);
+        FlagEvaluationDetails<Boolean> want = FlagEvaluationDetails.<Boolean>builder()
+                .value(true)
+                .variant("True")
+                .flagKey("unknown_field")
+                .reason(Reason.TARGETING_MATCH.name())
+                .flagMetadata(defaultMetadata)
+                .build();
+        assertEquals(want, got);
+        OpenFeatureAPI.getInstance().shutdown();
     }
 
     @SneakyThrows
     @Test
     void should_publish_events() {
-        GoFeatureFlagProvider g = new GoFeatureFlagProvider(GoFeatureFlagProviderOptions.builder().endpoint(this.baseUrl.toString()).timeout(1000).build());
-        g.initialize(new ImmutableContext());
-        g.getBooleanEvaluation("bool_targeting_match", false, this.evaluationContext);
-        g.getBooleanEvaluation("bool_targeting_match", false, this.evaluationContext);
-        g.getBooleanEvaluation("bool_targeting_match", false, this.evaluationContext);
-        assertEquals(0, g.getEventsPublisher().publish(), "first attempt expected to fail");
-
-        // simulate publish on next interval
-        assertEquals(2, g.getEventsPublisher().publish(), "expected to publish all events after retry");
```

Comment:
This test simulating a single failure, such that event is left in queue, and then published on next attempt with retry. Why removed ?